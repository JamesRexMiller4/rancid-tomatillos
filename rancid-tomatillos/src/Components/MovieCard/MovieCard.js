import React from 'react';
import './MovieCard.scss';
import images from '../../images/images';
import { Link } from 'react-router-dom';

const MovieCard = ({ id, poster, avgRating, userRating, isLoggedIn }) => {
  const determineIcon = (rating) => {
    if (rating < 4) {
      return images.badTomatillo;
    } else if (rating > 3 && rating < 7) {
      return images.okayTomatillo;
    } else if (rating > 6) {
      return images.goodTomatillo;
    }
  }
  let userRatingInfo;
  const roundedAvgRating = Math.ceil(avgRating);
  const avgRatingIcon = determineIcon(roundedAvgRating);
  if (userRating.length) {
    const userRatingIcon = determineIcon(userRating[0].rating);
    userRatingInfo = (
      <section className='section-avg-rating'>
        <p className='average-rating-p'>User Rating</p>
        <div className='rating-section'>
          <img src={userRatingIcon} alt='rancid tomatillo' className='tomatillo-icon' />
          <p className='average-rating-num'>{userRating[0].rating}</p>
        </div>
      </section>
    )
  } else {
    userRatingInfo = (
      <p className='user-no-rating-p'>No User Rating</p>
    )
  }
  return (
    <Link to={`/movies/${id}`} className='movie-detail-link'>
      <article data-id={id} className='movie-card'>
        <section className='section-movie-image' style={{backgroundImage: `url(${poster})`}}>
        </section>
        <div className='movie-rating-container'>
          <section className='section-avg-rating'>
            <p className='average-rating-p'>Average Rating</p>
            <section className='rating-section'>
              <img src={avgRatingIcon} alt='rancid tomatillo' className='tomatillo-icon' />
              <p className='average-rating-num'>{roundedAvgRating}</p>
            </section>
          </section>
          {isLoggedIn && userRatingInfo}
        </div>
      </article>
    </Link>
  );
}

export default MovieCard;
