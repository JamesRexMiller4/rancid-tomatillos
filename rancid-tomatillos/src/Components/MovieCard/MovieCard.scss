.movie-card {
  display: flex;
  align-items: flex-start;
  justify-content: space-between;
  width: 408px;
  height: 397px;
  border-radius: 10px;
  background-color: #4F776C;
  color: white;
  margin: 10px 0;
  border: 2px solid black;
  cursor: pointer;
}

.movie-card:hover {
  box-shadow: 5px 10px rgba(0, 0, 0, .6);

}

.section-movie-image {
  width: 65%;
  height: 100%;
  border-radius: 10px;
  background-size: cover;
}

.section-avg-rating {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 35%;
  height: 50%;
  text-align: center;
}

.average-rating-p {
  font-size: 1.5em;
  margin: 0;
  width: 80%;
  padding: 5px;
  border-bottom: 2px solid black;
}

.rating-section {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
}

.tomatillo-icon {
  height: 50px;
  width: 50px;
}

.average-rating-num {
  font-size: 2.5em;
  margin: 20px 10px;;
}
