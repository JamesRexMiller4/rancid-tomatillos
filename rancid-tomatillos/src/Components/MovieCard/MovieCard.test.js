import React from 'react';
import {shallow} from 'enzyme';
import MovieCard from './MovieCard';
 
describe('Card', () => {
let wrapper;
 
beforeEach(() => {
wrapper = shallow(<MovieCard />)
})
 
it.skip('', () => {
})
 
});