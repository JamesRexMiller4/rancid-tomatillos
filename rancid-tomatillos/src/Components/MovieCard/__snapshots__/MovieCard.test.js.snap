// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard component should match the snapshot while the user is loggedIn and there are userRatings 1`] = `
<Link
  className="movie-detail-link"
  to="/movies/1"
>
  <article
    className="movie-card"
    data-id={1}
  >
    <section
      className="section-movie-image"
      style={
        Object {
          "backgroundImage": "url(https://image.tmdb.org/t/p/original//l4iknLOenijaB85Zyb5SxH1gGz8.jpg)",
        }
      }
    />
    <div
      className="movie-rating-container"
    >
      <section
        className="section-avg-rating"
      >
        <p
          className="average-rating-p"
        >
          Average Rating
        </p>
        <section
          className="rating-section"
        >
          <img
            alt="rancid tomatillo"
            className="tomatillo-icon"
            src="goodTomatillo.png"
          />
          <p
            className="average-rating-num"
          >
            7
          </p>
        </section>
      </section>
      <section
        className="section-avg-rating"
      >
        <p
          className="average-rating-p"
        >
          User Rating
        </p>
        <div
          className="rating-section"
        >
          <img
            alt="rancid tomatillo"
            className="tomatillo-icon"
          />
          <p
            className="average-rating-num"
          />
        </div>
      </section>
    </div>
  </article>
</Link>
`;

exports[`MovieCard component should match the snapshot while the user is loggedIn but with no userRatings 1`] = `
<Link
  className="movie-detail-link"
  to="/movies/1"
>
  <article
    className="movie-card"
    data-id={1}
  >
    <section
      className="section-movie-image"
      style={
        Object {
          "backgroundImage": "url(https://image.tmdb.org/t/p/original//l4iknLOenijaB85Zyb5SxH1gGz8.jpg)",
        }
      }
    />
    <div
      className="movie-rating-container"
    >
      <section
        className="section-avg-rating"
      >
        <p
          className="average-rating-p"
        >
          Average Rating
        </p>
        <section
          className="rating-section"
        >
          <img
            alt="rancid tomatillo"
            className="tomatillo-icon"
            src="goodTomatillo.png"
          />
          <p
            className="average-rating-num"
          >
            7
          </p>
        </section>
      </section>
      <p
        className="user-no-rating-p"
      >
        No User Rating
      </p>
    </div>
  </article>
</Link>
`;

exports[`MovieCard component should match the snapshot with all the data being passed in correctly while not logged in 1`] = `
<Link
  className="movie-detail-link"
  to="/movies/1"
>
  <article
    className="movie-card"
    data-id={1}
  >
    <section
      className="section-movie-image"
      style={
        Object {
          "backgroundImage": "url(https://image.tmdb.org/t/p/original//l4iknLOenijaB85Zyb5SxH1gGz8.jpg)",
        }
      }
    />
    <div
      className="movie-rating-container"
    >
      <section
        className="section-avg-rating"
      >
        <p
          className="average-rating-p"
        >
          Average Rating
        </p>
        <section
          className="rating-section"
        >
          <img
            alt="rancid tomatillo"
            className="tomatillo-icon"
            src="goodTomatillo.png"
          />
          <p
            className="average-rating-num"
          >
            7
          </p>
        </section>
      </section>
    </div>
  </article>
</Link>
`;
